NebulaWorld/SimulatedWorld.cs

```diff
@@ -286,25 +353,28 @@ public static void OnDronesDraw()
         {
             foreach(KeyValuePair<ushort, RemotePlayerModel> remoteModel in remotePlayersModels)
             {
-                remoteModel.Value.MechaInstance.droneRenderer.Draw();
+                //Render drones of players only on the local planet
+                if (GameMain.mainPlayer.planetId == remoteModel.Value.Movement.localPlanetId)
+                {
+                    remoteModel.Value.MechaInstance.droneRenderer.Draw();
+                }
             }
         }
 
         public static void OnDronesGameTick(long time, float dt)
         {
-            double tmp = 0;
-            double tmp2 = 0;
+            double tmp = 1e10; //fake energy of remote player, needed to do the Update()
+            double tmp2 = 1;
             //Update drones positions based on their targets
             foreach (KeyValuePair<ushort, RemotePlayerModel> remoteModel in remotePlayersModels)
             {
                 MechaDrone[] drones = remoteModel.Value.PlayerInstance.mecha.drones;
-                int droneCount = remoteModel.Value.PlayerInstance.mecha.droneCount;
+                int droneCount = GameMain.mainPlayer.mecha.droneCount;
                 for (int i = 0; i < droneCount; i++)
```

Comment:
This feels wrong to me, you iterate over the remote players drones by using the local player drones count ??

Reply:
Well is the drones count the count of available or active drones? If available, the tech level is synced, so the count should be the same.

Reply:
It seemed to be unnecessary to sync also `droneCount` for every remote model since this value is the same for all players. This number represents how many drones the mecha has which is the same for all players.
If I would iterate over the remote player's drones count I would also have to implement synching of a number of drones for remote mechas everything someone joins / every time some research unlocks it. And for me, it was an unnecessary step.

Reply:
Ok yeah that's good then.